src/frontend/components/App.jsx

```diff
@@ -14,20 +16,81 @@ const ThankYou = lazy(() => import('./ThankYou.jsx'));
 const Survey = lazy(() => import('./Survey.jsx'));
 
 export default function App() {
-  return (
-    <Container>
-      <ErrorBoundary FallbackComponent={Error}>
-        <Switch>
-          <LazyBoundary fallback={Loading}>
-            <Route exact path="/" render={props => <Front {...props} />} />
-            <Route path="/login" render={props => <Login {...props} />} />
-            <Route path="/admin" render={props => <Dashboard {...props} />} />
-            <Route path="/thankyou" render={props => <ThankYou {...props} />} />
-            <Route path="/survey" render={props => <Survey {...props} />} />
-          </LazyBoundary>
-          <Redirect to="/" />
-        </Switch>
-      </ErrorBoundary>
-    </Container>
-  );
+  const [user, setUser] = React.useState(null);
+  const [error, setError] = React.useState(null);
+
+  const processError = res => {
+    let errorString;
+    if (res.statusCode && res.error && res.message) {
+      errorString = `HTTP ${res.statusCode} ${res.error}: ${res.message}`;
+    } else if (res.statusCode && res.status) {
+      errorString = `HTTP ${res.statusCode}: ${res.status}`;
+    } else if (res.message) {
+      errorString = res.message;
+    } else {
+      errorString = 'Error in response from server.';
+    }
+    return errorString;
+  };
+
+  // fetch api data
+  React.useEffect(() => {
+    let isMounted = true;
+    let userStatus;
+    const username = Cookies.get('p_user');
```

Comment:
See the comment above about limiting access to endpoints. Currently the tabs are hidden according to the username, but if I edit my `p_user` cookie to be 'admin' I have admin access. At the very least we should limit access on the backend to `POST` endpoints for settings and forms.